app/uk/gov/hmrc/apipublisher/models/APIApproval.scala

```diff
@@ -27,7 +27,7 @@ import uk.gov.hmrc.apipublisher.models.ApprovalStatus.APPROVED
 case class ApiApprovalState(
     actor: Actor,
     changedAt: Instant,
-    status: ApprovalStatus,
+    status: Option[ApprovalStatus] = None,
```

Comment:
Is it possible to avoid a none and where there is a none on insert of comments make it the previous value?  Or if it is necessary then perhaps a trait/case hierachy matching the 4 (or so) operations - basically a kind of event/command structure?

Reply:
Before this change it was doing just that i.e. setting the status to the previous value on insert of comment. 
This caused an issue on GK with not being able to tell the difference  between an actual status change and just a comment add.
We did discuss using a trait/case hierarchy but felt there were only two cases so this seemed the simpler solution.

Reply:
Ok